File: server/api/types/connection/world-update.go
package connection

import (
	"github.com/dayaftereh/discover/server/mathf"
)

type GameObject struct {
	Type       *string     `json:"type"`
	Radius     *float64    `json:"radius"`
	Position   *mathf.Vec3 `json:"position"`
	Rotation   *mathf.Vec3 `json:"rotation"`
	Removeable *bool       `json:"removeable"`
	Color      *int64      `json:"color"`
}

type PlayerUpdate struct {
	GameObjectId *int64 `json:"gameObjectId"`
}

type WorldUpdateMessage struct {
	Type    MessageType           `json:"type"`
	Tick    *int64                `json:"tick"`
	Time    *float64              `json:"time"`
	Player  *PlayerUpdate         `json:"player"`
	Objects map[int64]*GameObject `json:"objects"`
}

File: app/src/app/services/api/connection/messages/world-update.ts
import { Message } from "./message";

export interface Vec3 {
    x: number
    y: number
    z: number
}

export interface GameObject {
    type?:string
    radius?: number
    position?: Vec3
    rotation?: Vec3
    removeable?: boolean
    color?: number
}

export interface PlayerUpdate {
    gameObjectId: number
}

export interface WorldUpdate extends Message {
    tick: number
    time: number
    player: PlayerUpdate
    objects: { [key: number]: GameObject }
}